src/index.ts

```diff
@@ -791,12 +791,11 @@ export class BufferedChangeset implements IChangeset {
    */
   _valueFor(key: string): any {
     let changes: Changes = this[CHANGES];
-    let errors: Errors<any> = this[ERRORS];
     let content: Content = this[CONTENT];
 
-    if (errors.hasOwnProperty(key)) {
-      let e: Err = errors[key];
-      return e.value;
```

Comment:
What are your thoughts on removing this?

Reply:
I've put it in a separate commit with my thoughts.
Basically I think that this "caching" is redundant here and can be error-prone.
I initially put the logic of taking the property after that section and it didn't work because the cache wasn't invalidated when related fields were changed. So, I've put it before that block and then removed this `if` section in a separate to get your feedback about that.

If you think it is useful, I can just drop the last commit.

Reply:
Gotcha!  Looks like we don't have a test for this case.  If we try and validate a key, is it possible the validation results only lives in `ERRORS`?  Perhaps we can add a test case to check this assumption?

Reply:
The code I removed just accesses the value, contained in `ERRORS` object, not the validation results. The value can be always accessed from the `changes` or the underlying object. Therefore I consider this thing as a performance optimization, not the necessary functionality.

Maybe I'm missing something?

In any case, this change is not really tied to the original issue I'm trying to address. I can extract it to a separate PR or just drop it.

What do you think about the change in `30c0b2a`?

Reply:
Separate PR may be good unless we can confirm with a test.  The changes you have for the specific issue seem 💯

Reply:
> value can be always accessed from the changes or the underlying object

That is - a test confirming that adding an error will result in the right validations.  (perhaps we already have one)

Reply:
If you mean `addError` API, there is a bunch of tests tagged with #addError.